qiskit/algorithms/optimizers/suffix_averaging.py

```diff
@@ -0,0 +1,154 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Suffix Averaging optimier"""
+
+import warnings
+from typing import Callable, List, Optional, Tuple
+
+import numpy as np
+
+from qiskit.algorithms.optimizers import NFT, QNSPSA, SPSA, GradientDescent
+
+from .optimizer import POINT, Optimizer, OptimizerResult
+
+
+class SuffixAveragingOptimizer(Optimizer):
+    r"""The suffix averaging optimizer.
+
+    Given a sequence of circuit parameters of parameterized quantum circuit
+    obtained from :math:`T` iterations, :math:`\{\vec{\theta}^{(t)}\}_{t=1}^{t=T}`,
+    the SuffixAveragingOptimizer returns the averaged point of the last n_params_suffix
+    points,
+
+    .. math::
+
+        \overline{\vec{\theta}} = \frac{1}{n_params_suffix} \sum_{t=T-n_params_suffix-1}^{T}
+        \vec{\theta}^{(t)}.
+
+    Examples:
+        .. code-block::python
+            from qiskit import Aer
+            from qiskit.utils import QuantumInstance
+            from qiskit.algorithms import VQE
+            from qiskit.opflow.gradients import Gradient
+            from qiskit.circuit.library import TwoLocal
+            from qiskit.algorithms.optimizers import ADAM
+            from qiskit.algorithms.optimizers.suffix_averaging import SuffixAveragingOptimizer
+
+            backend = Aer.get_backend("qasm_simulator")
+            quantum_instance = QuantumInstance(backend=backend, shots=1000)
+            ansatz = TwoLocal(4, ["rx", "rz"], "cx", "linear")
+            optimizer = ADAM(maxiter=1000)
+            suffix_optimizer = SuffixAveragingOptimizer(optimizer, n_params_suffix=50)
+            gradient = Gradient(grad_method="param_shift")
+            vqe = VQE(
+                ansatz = ansatz,
+                optimizer = suffix_optimizer,
+                gradient = gradient,
+                quantum_instance = quantum_instance
+            )
+
```

Comment:
I think we can drop this complete example no? Its fairly straightforward to use it is it not? Reason for saying this is that the algorithms are about to undergo a refactor to using primitives ahead of the next release and the above will end up being deprecated usage.

Reply:
Thank you for the guideline @woodsp-ibm. I removed it in [cead94d](https://github.com/Qiskit/qiskit-terra/pull/8628/commits/cead94debba9b3ca1d0a6ff226c07d9008218a41).